File: LeetCode/DataStructure/C_Plus_Plus/1.cpp
//https://leetcode.com/problems/two-sum/

class Solution
{
    public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        int diff;
        unordered_map<int,int> hash_map;
        for ( int i = 0; i < nums.size(); i++ )
        {
            diff = target - nums[i];
            if ( hash_map.find(nums[i]) == hash_map.end() )
            {
                hash_map[diff] = i;
            }
            else
            {
                return {hash_map[nums[i]], i};
            }
        }
        return {-1,-1};
    }
};

File: LeetCode/Study_Plan/DataStructure/Python/1.py
'''https://leetcode.com/problems/two-sum/'''

'''Using Hashing Technique'''
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        preVal = {}
        for index , data in enumerate(nums):
            diff = target - data
            if diff in preVal:
                return [preVal[diff],index]
            preVal[data] = index
        return -1